coachmark/src/main/java/com/pseudoankit/coachmark/overlay/OverlayLayout.kt

```diff
@@ -0,0 +1,147 @@
+package com.pseudoankit.coachmark.overlay
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.UiComposable
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.layout.layoutId
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.pseudoankit.coachmark.model.ToolTipPlacement
+import com.pseudoankit.coachmark.model.TooltipConfig
+
+/** Containing composable must use these values for layoutId on current and previous tooltip. */
+public enum class OverlayChildLayoutId { CURRENT, PREVIOUS }
+
+/** Proposed minimum useful tooltip width; this can be adjusted if needed */
+private const val TOOLTIP_MAX_WIDTH_OVERRIDE_PX = 30
+
+/**
+ * Extracted from DimOverlayEffect, in case another overlay effect needs it in the future.
+ *
+ * @param configCurrent of coach mark highlight; null suits case where there's no "current" tooltip in content
+ * @param configPrevious of coach mark highlight; null suits case where there's no "previous" tooltip in content
+ * @param gapTooltipScreen min distance between tooltip and edge of screen
+ */
+@Composable
+public fun OverlayLayout(
```

Comment:
can you please push the changes if done, I am not seeing it yet

Reply:
I see your response on the enum issue, but we should settle on a replacement for gapTooltipScreen.  I think no name is perfect; if you want to make the final call, I should be able to push tomorrow morning.